libraries/AP_Mission/AP_Mission_Relative.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * AP_Mission_Relative.h
+ *
+ *  Created on: 08.05.2020
+ *      Author: WillyZehnder
+ */
+/// @file    AP_Mission_Relative.h
+/// @brief   translates and rotates Missions according to the location where Mode-AUTO has been switched on
+
+/*
+ *   The AP_Mission_Relative library:
+ *   - memorizes the Location where Mode-AUTO has been switched on (Base-Point)
+ *   - moves the individual Waypoints according to Base-Point and Parameters
+ */
+#pragma once
+
+#ifndef ENABLED
+    #define ENABLED 1
+#endif
+
+#include <AP_Param/AP_Param.h>
+#include <AP_Vehicle/ModeReason.h>
+
+// definitions
+#ifndef AP_MISSION_RELATIVE_NO_TRANSLATION_RADIUS_DEFAULT
+    // no translation of the Mission within a radius of XX m around Homepoint (it's a compromise: for Copter/Rover/Boat 5-10m, for Plane 50-100m should be sufficient)
+    #define AP_MISSION_RELATIVE_NO_TRANSLATION_RADIUS_DEFAULT 50
+#endif
+#define AP_MISSION_RELATIVE_KIND_OF_MOVE_DEFAULT    0   // no translation
+#define AP_MISSION_RELATIVE_OPTIONS_DEFAULT         0   // don't use altitude offset
+
+    class AP_Mission;
+
+/// @class    AP_Mission_Relative
+/// @brief    Object managing movement of Waypoints
+class AP_Mission_Relative
+{
+
+    friend class AP_Mission;
+
+public:
+
+    AP_Mission_Relative(void);
+
+    /* Do not allow copies */
+    AP_Mission_Relative(const AP_Mission_Relative &other) = delete;
+    AP_Mission_Relative &operator=(const AP_Mission_Relative&) = delete;
+
+    // user settable parameters
+    static const struct AP_Param::GroupInfo var_info[];
+
+    static AP_Mission_Relative *get_singleton(void) {
+        return _singleton;
+    }
+
+private:
+
+    static AP_Mission_Relative *_singleton;
+
+    AP_Int8     _kind_of_move;          // defines the kind of move (translation/rotation) of the mission when entering Auto mode
+    AP_Int8     _rel_options;           // bitmask options for special behaviour at translation
+    AP_Float    _no_translation_radius; // distance in [m] from HomeLocation wherein a translation of a Relative Mission will be ignored
+    AP_Int8     _rotate_ch;             // channel to rotate by rc-channel-input
+
+    AP_Int8     _rel_options_fixed;     // bitmask options for special behaviour at translation - fixed at Mission-Start
+
+    enum class Restart_Behaviour {
+        RESTART_NOT_TRANSLATED,
```

Comment:
If you define the enum values here, and document them. You can use them in the Restart_Behaviour swicht case and simplify the code.

Reply:
Sorry, I can't see where you're pointing to - I still use them in the switch:
`        // calculate amount of rotation
        switch (_restart_behaviour) {
            case Restart_Behaviour::RESTART_ROTATED_NORTH:
            case Restart_Behaviour::RESTART_ROTATED_FIRST_WP:
                tmp.content.location = AP::ahrs().get_home();
                // get direction from Homepoint to Basepoint (North-related)
                _translation.direction = tmp.content.location.get_bearing_to(_basepoint_loc);   // in centidegrees from 0 36000
                break;
            case Restart_Behaviour::RESTART_ROTATED_HEADING:
                // rotation via heading at Basepoint
                _translation.direction = AP::ahrs().yaw_sensor;
                break;`